tgui/packages/tgui/interfaces/Atm/index.tsx

```diff
@@ -0,0 +1,182 @@
+/**
+ * @file
+ * @copyright 2022
+ * @author DisturbHerb (https://github.com/DisturbHerb/)
+ * @license MIT
+ */
+
+import { useBackend, useLocalState } from '../../backend';
+import { Box, Button, Divider, Icon, NoticeBox, Section, Tabs } from '../../components';
+import { Window } from '../../layouts';
+import { AtmData, AtmTabKeys } from './types';
+
+export const Atm = (_, context) => {
+  const { data } = useBackend<AtmData>(context);
+  const { name } = data;
+
+  const [tabIndex, setTabIndex] = useLocalState(context, 'tabIndex', AtmTabKeys.Teller);
+  return (
+    <Window title={name} width={375} height={450}>
+      <Window.Content>
+        <Tabs fluid>
+          <Tabs.Tab
+            icon="money-bills"
+            selected={tabIndex === AtmTabKeys.Teller}
+            onClick={() => setTabIndex(AtmTabKeys.Teller)}>
+            ATM
+          </Tabs.Tab>
+          <Tabs.Tab
+            icon="coins"
+            selected={tabIndex === AtmTabKeys.Spacebux}
+            onClick={() => setTabIndex(AtmTabKeys.Spacebux)}>
+            Spacebux
+          </Tabs.Tab>
+        </Tabs>
+        {tabIndex === AtmTabKeys.Teller && <Teller />}
+        {tabIndex === AtmTabKeys.Spacebux && <SpacebuxMenu />}
+      </Window.Content>
+    </Window>
+  );
+};
+
+const Types = {
+  Danger: 'danger',
+  Info: 'info',
+  Success: 'success',
+};
+
+const TypedNoticeBox = (props) => {
+  const { type, ...rest } = props;
+  const typeProps = {
+    ...(type === Types.Danger ? { danger: true } : {}),
+    ...(type === Types.Info ? { info: true } : {}),
+    ...(type === Types.Success ? { success: true } : {}),
+  };
+  return <NoticeBox {...typeProps} {...rest} />;
+};
```

Comment:
why not just add the types to the actual notice box?

Reply:
IDE got angy if i tried, this is how it's implemented in the cloning console as well, would love to hear how you'd fix it